modules/caddyhttp/reverseproxy/reverseproxy.go

```diff
@@ -129,8 +129,12 @@ func (h *Handler) Provision(ctx caddy.Context) error {
 	h.ctx = ctx
 	h.logger = ctx.Logger(h)
 
+	re := regexp.MustCompile(`\{[[:graph:]]+\}`)
```

Comment:
Not gonna lie, I do hate this :laughing: Let me think on this for a bit and see if we can come up with something more elegant...

This didn't used to be a problem, right? Did we _add_ code that broke this use case? Can we not remove that code? Or at least change it?

Reply:
Yeah, the code added to fix the healthcheck on alternate port requires parsing the network address, which errors out for not having a port when using placeholders. I've moved that part into later in the `Provision` method where it only sets up the health checks. Now if the upstream contains a placeholder dial address, it will not accept active healthcheck config because it's like SRV and owned by another entity.

Reply:
Do you think there's a way we can fix the original health checking bug without requiring host/port information up front? That would simplify the code, rather than complicate it.